phoenix-core/src/main/java/org/apache/phoenix/index/IndexMetaDataCacheClient.java

```diff
@@ -136,7 +137,11 @@ public static ServerCache setMetaDataOnMutations(PhoenixConnection connection, P
             }
             mutation.setAttribute(PhoenixIndexCodec.INDEX_UUID, uuidValue);
             if (table.getTransformingNewTable() != null) {
-                mutation.setAttribute(BaseScannerRegionObserver.DO_TRANSFORMING, TRUE_BYTES);
+                boolean disabled = table.getTransformingNewTable().getIndexState()!= null &&
```

Comment:
since this is a very frequent check across several classes, should it be encapsulated as a function somewhere? (on PTable or in a Util class?)